scripts/prepdocslib/textsplitter.py

```diff
@@ -16,16 +18,92 @@ def split_pages(self, pages: List[Page]) -> Generator[SplitPage, None, None]:
             yield  # pragma: no cover - this is necessary for mypy to type check
 
 
+STANDARD_LINE_BREAKS = [",", ";", ":", " ", "(", ")", "[", "]", "{", "}", "\t", "\n"]
+CJK_LINE_BREAKS = [
+    "。",
+    "、",
+    "，",
+    "；",
+    "：",
+    "（",
+    "）",
+    "【",
+    "】",
+    "「",
+    "」",
+    "『",
+    "』",
+    "〔",
+    "〕",
+    "〈",
+    "〉",
+    "《",
+    "》",
+    "〖",
+    "〗",
+    "〘",
+    "〙",
+    "〚",
+    "〛",
+    "〝",
+    "〞",
+    "〟",
+    "〰",
+    "〾",
+    "〿",
+    "–",
+    "—",
+    "‘",
+    "’",
+    "‚",
+    "‛",
+    "“",
+    "”",
+    "„",
+    "‟",
+    "‹",
+    "›",
+    "‽",
+    "⁇",
+    "⁈",
+    "⁉",
+    "⸮",
+    "ⸯ",
+    "。",
+    "々",
+    "〻",
+    "〼",
+]
+cl100k = tiktoken.get_encoding("cl100k_base")
```

Comment:
This should also be configurable incase there is another BPE in the future?

Reply:
The repo has an existing incompatibility with ollama and local language models as our token-counting code in https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/app/backend/core/modelhelper.py assumes OpenAI models. I see here that you assume cl100k_base versus using encoding_for_model(). On the one hand, that means that this code will work for any model, but that also means it may work deceptively. (i.e. it thinks its tokenizing correctly, but it isn't).
On the other hand:
* the current azure openai models all use that BPE- https://github.com/openai/tiktoken/blob/1b9faf2779855124f05174adf1383e53689ed94b/tiktoken/model.py#L19
* supporting non-openai models would require bringing in the tokenizers library so we're far out from that.

I think my vote is to use tiktoken.encoding_for_model() with the name of the embedding model, since then people would realize the error with using non-openai models. I believe prepdocs already has access to the model name somewhere.